src/format/parse.rs

```diff
@@ -424,25 +424,16 @@ where
                     }
 
                     &Internal(InternalFixed { val: InternalInternal::Nanosecond3NoDot }) => {
-                        if s.len() < 3 {
```

Comment:
Can you explain in the commit message why they're duplicate (that is, how these invariants are upheld)? It's not clear in the diff.